cadcdata/cadcdata/version.py

```diff
@@ -0,0 +1,217 @@
+# Autogenerated by Astropy-affiliated package cadcdata's setup.py on 2016-12-14 11:36:19.416397
+from __future__ import unicode_literals
+import datetime
+
+
+import locale
+import os
+import subprocess
+import warnings
+
+
+def _decode_stdio(stream):
+    try:
+        stdio_encoding = locale.getdefaultlocale()[1] or 'utf-8'
+    except ValueError:
+        stdio_encoding = 'utf-8'
+
+    try:
+        text = stream.decode(stdio_encoding)
+    except UnicodeDecodeError:
+        # Final fallback
+        text = stream.decode('latin1')
+
+    return text
+
+
+def update_git_devstr(version, path=None):
+    """
+    Updates the git revision string if and only if the path is being imported
+    directly from a git working copy.  This ensures that the revision number in
+    the version string is accurate.
+    """
+
+    try:
+        # Quick way to determine if we're in git or not - returns '' if not
+        devstr = get_git_devstr(sha=True, show_warning=False, path=path)
+    except OSError:
+        return version
+
+    if not devstr:
+        # Probably not in git so just pass silently
+        return version
+
+    if 'dev' in version:  # update to the current git revision
+        version_base = version.split('.dev', 1)[0]
+        devstr = get_git_devstr(sha=False, show_warning=False, path=path)
+
+        return version_base + '.dev' + devstr
+    else:
+        # otherwise it's already the true/release version
+        return version
+
+
+def get_git_devstr(sha=False, show_warning=True, path=None):
+    """
+    Determines the number of revisions in this repository.
+
+    Parameters
+    ----------
+    sha : bool
+        If True, the full SHA1 hash will be returned. Otherwise, the total
+        count of commits in the repository will be used as a "revision
+        number".
+
+    show_warning : bool
+        If True, issue a warning if git returns an error code, otherwise errors
+        pass silently.
+
+    path : str or None
+        If a string, specifies the directory to look in to find the git
+        repository.  If `None`, the current working directory is used, and must
+        be the root of the git repository.
+        If given a filename it uses the directory containing that file.
+
+    Returns
+    -------
+    devversion : str
+        Either a string with the revision number (if `sha` is False), the
+        SHA1 hash of the current commit (if `sha` is True), or an empty string
+        if git version info could not be identified.
+
+    """
+
+    if path is None:
+        path = os.getcwd()
+        if not _get_repo_path(path, levels=0):
+            return ''
+
+    if not os.path.isdir(path):
+        path = os.path.abspath(os.path.dirname(path))
+
+    if sha:
+        # Faster for getting just the hash of HEAD
+        cmd = ['rev-parse', 'HEAD']
+    else:
+        cmd = ['rev-list', '--count', 'HEAD']
+
+    def run_git(cmd):
+        try:
+            p = subprocess.Popen(['git'] + cmd, cwd=path,
```

Comment:
Good comment for the astropy dev team. If I understand it correctly, we shouldn't run that code since we are not using 'dev' in our revision numbers.